odbc/src/api/data.rs

```diff
@@ -115,6 +115,22 @@ impl IntoCData for Bson {
             Bson::Int64(i) => Ok(i.to_le_bytes().to_vec()),
             Bson::Binary(b) => Ok(b.bytes),
             Bson::Decimal128(d) => Ok(d.bytes().to_vec()),
+            Bson::Document(d) => Ok(d.to_string().into_bytes()),
+            Bson::Array(a) => Ok(serde_json::to_string(
+                &a.into_iter().map(|b| b.to_json()).collect::<Vec<String>>(),
+            )
+            .unwrap()
+            .into_bytes()),
+            Bson::ObjectId(o) => Ok(o.to_string().into_bytes()),
+            Bson::Undefined => Ok(char::from(0).to_string().into_bytes()),
+            Bson::Null => Ok(char::from(0).to_string().into_bytes()),
+            Bson::RegularExpression(r) => Ok(r.to_string().into_bytes()),
+            Bson::DbPointer(_) => Ok(char::from(0).to_string().into_bytes()),
```

Comment:
For `DbPointer` there is a value we can display.  See if something like this works:
```
Bson::DbPointer(d) => Ok(serde_json::to_string(&d).unwrap().into_bytes()),
```